### PR TITLE
Add a solution for Homework 3

### PR DESCRIPTION
**Pull request by tzurowski**
Tests are positive, now it's time for some criticism from mentors.

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

File: Src/BootCamp.Chapter/Program.cs [hunk 3]
```diff
@@ -1,14 +1,83 @@
...
+        }
+        public static string InputString(string message)
+        {
+            Console.WriteLine(message);
+            //Convert to string
+            string input = Console.ReadLine();
+            return input;
+        }
+       
+        public static float CalculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

File: Src/BootCamp.Chapter/lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class lesson3
```
Comment (Almantask) [inline Src/BootCamp.Chapter/lesson3.cs]: Codacy Issue found: [Rename class 'lesson3' to match pascal case naming rules, consider using 'Lesson3'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson31.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/Lesson31.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson31.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

File: Src/BootCamp.Chapter/Lesson31.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson31.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson31.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/Lesson31.cs (hunk 5 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson31.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

File: Src/BootCamp.Chapter/Lesson31.cs (hunk 6 above)
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson31.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/lesson3.cs (hunk 4 above)
Comment (ellmaster) [inline Src/BootCamp.Chapter/lesson3.cs]: Class and file name should be written with a capital L.

File: Src/BootCamp.Chapter/Program.cs [hunk 7]
```diff
@@ -1,14 +1,83 @@
...
+        {
+            //Calculate BMI
+            float bodyMassIndex = weight / (float)Pow(height, 2);
+            return bodyMassIndex;
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Program.cs]: Try to return without caching the result first. You don't need that variable. If Codacy blames you for doing so just ignore it.

File: Src/BootCamp.Chapter/lesson3.cs (hunk 4 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter/lesson3.cs]: I corrected it earlier, could you check on recent commits?

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/Program.cs (hunk 7 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter/Program.cs]: I added few changes like you advised me, u can check the latest version.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/Checks.cs [hunk 8]
```diff
@@ -17,25 +13,26 @@ public static class Checks
         public static int PromptInt(string message)
         {
             // To do: call your implementation.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove redundant comments :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            string decision;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Well done for moving to if clauses :)

File: Src/BootCamp.Chapter/Program.cs [hunk 10]
```diff
@@ -1,14 +1,75 @@
...
-using System.Collections.Generic;
-using System.IO;
+using static System.Math;
 using Console = System.Console;
 
 namespace BootCamp.Chapter
 {
     class Program
     {
+        /*  Homework 3
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Move this comment as an xml summary above either Lesson3 class or Demo method. Like that, hovering your mouse over that method will show its' description. You don't want to force it in code, it messes up the readability.

File: Src/BootCamp.Chapter/Program.cs [hunk 11]
```diff
@@ -1,14 +1,75 @@
...
+                - Prompt for input and converting it to float. 
+            3) Put all the function you made into the right places in Checks
+            4) Move all the code from main function into lesson3 class function called Demo. Call it from main function.
+        */
+
         static void Main(string[] args)
         {
+            Lesson3.Demo();
+        }
+        public static string AddNewPerson()
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Leave program only with the main method. All other methods have a better place- which is the Lesson class.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

File: Src/BootCamp.Chapter/Checks.cs (hunk 8 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter/Checks.cs]: Done

File: Src/BootCamp.Chapter/Program.cs (hunk 10 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter/Program.cs]: Done

File: Src/BootCamp.Chapter/Program.cs (hunk 11 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter/Program.cs]: Done

File: Src/BootCamp.Chapter/Program.cs [hunk 12]
```diff
@@ -1,75 +1,22 @@
...
+/// 1) Fork https://github.com/csinn/CSharp-From-Zero-To-Hero/tree/Chapter1/Homework/3 
+/// 2) Take homework 2 code, place it in the forked branch and refactor it using functions. There should be as little duplicate code as possible(there should be functions for: 
+/// - Calculating BMI,
+/// - Prompt for input and converting it to int,
+/// - Prompt for input and converting it to string,
+/// - Prompt for input and converting it to float. 
+/// 3) Put all the function you made into the right places in Checks
+/// 4) Move all the code from main function into lesson3 class function called Demo.Call it from main function.
+/// </summary>
+/// <param name="args"></param>
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Program.cs]: You want this to be of the same indention as the thing you are commenting.
So for classes have it on class level indention and for functions on function level indention. :)

Comment (tzurowski) [reply Src/BootCamp.Chapter/Program.cs]: Just click CTRL + K + D but ok :P

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845544&bid=16083462)

**Pull request closed without merging**